tldw_Server_API/app/core/DB_Management/chacha/message_store.py

```
@@ -0,0 +1,925 @@
+                return None
+            return dict(row) if isinstance(row, dict) else {
+                "id": row[0],
+                "timestamp": row[1],
+                "content": row[2],
+                "sender": row[3],
+            }
+        except CharactersRAGDBError as exc:
+            logger.error("Database error fetching latest message for conversation {}: {}", conversation_id, exc)
+            raise
+
+    def count_messages_since(
```

Ruff (0.15.10)

[warning] Too many return statements (7 > 6)

(PLR0911)